akka-http-tests/src/test/scala/akka/http/javadsl/server/HttpAppSpec.scala

```diff
@@ -22,48 +23,55 @@ import scala.runtime.BoxedUnit
 class HttpAppSpec extends AkkaSpec with RequestBuilding with Eventually {
   import system.dispatcher
 
-  "HttpApp" should {
+  "HttpApp  Java" should {
 
     "start without ActorSystem" in {
-      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
 
       val minimal = new MinimalHttpApp()
 
       val server = Future {
-        minimal.startServer(host, port, ServerSettings.create(ConfigFactory.load))
+        minimal.startServer("localhost", 0, ServerSettings.create(ConfigFactory.load))
```

Comment:
I think now I know why it failed that time, and the reason behind the deadlock in win.
It's not port collision (the stack trace we saw was the expected from binding failure test), it is that the server in rare occasions can't start in 5 seconds (start the actor system and the server). This causes an assertion failure, which prevents the server to be correctly shutdown, which is waiting for a `Future` to complete.
I could go back to the random port utility as it's simpler